openwisp_controller/notifications/handlers.py

```diff
@@ -0,0 +1,28 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from openwisp_notifications.signals import notify
+from swapper import load_model
+
+from openwisp_controller.config.signals import config_status_changed
+
+Config = load_model('config', 'Config')
+Device = load_model('config', 'Device')
+
+
+@receiver(
+    config_status_changed, sender=Config, dispatch_uid='config_change_notification'
+)
+def config_change_notification(sender, instance, **kwargs):
+    if instance.status == 'error':
+        notification_type = 'config_problem'
+    elif instance.status == 'applied':
+        notification_type = 'config_recovery'
```

Comment:
this looks naive, it looks like this will be triggered every time the config passes from modified to applied, which shouldn't happen, don't we already have the info of the previous status in the arguments?

Reply:
here's the issue with this one:

- when an error in the config happens, the user will debug, will find the cause of the issue and correct the configuration
- the config status will move from error to modified
- then the config is applied and status move into applied

So the status usually moves in this way:

- from modified to applied or viceversa
- from applied to error
- from error to modified

It should not happen that it moves from error to applied.

So, it's not really possible to easily know if an error state has recovered, we should dedicate more work to this to implement such a thing but I think it's too early now.

I think we could just leave this, we can focus on the error state and on the registration, for those two cases a notification is useful. In this case instead, we risk to generate too many notifications which will be counter productive.